commandapi-platforms/commandapi-spigot/commandapi-spigot-core/src/main/java/dev/jorel/commandapi/CommandAPISpigot.java

```diff
@@ -0,0 +1,137 @@
+package dev.jorel.commandapi;
+
+import dev.jorel.commandapi.commandsenders.BukkitBlockCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitConsoleCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitEntity;
+import dev.jorel.commandapi.commandsenders.BukkitNativeProxyCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitPlayer;
+import dev.jorel.commandapi.commandsenders.BukkitProxiedCommandSender;
+import dev.jorel.commandapi.commandsenders.BukkitRemoteConsoleCommandSender;
+import dev.jorel.commandapi.exceptions.WrapperCommandSyntaxException;
+import dev.jorel.commandapi.nms.NMS;
+import dev.jorel.commandapi.nms.SpigotNMS;
+import dev.jorel.commandapi.wrappers.NativeProxyCommandSender;
+import net.md_5.bungee.api.chat.BaseComponent;
+import org.bukkit.Bukkit;
+import org.bukkit.command.BlockCommandSender;
+import org.bukkit.command.CommandMap;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.ConsoleCommandSender;
+import org.bukkit.command.ProxiedCommandSender;
+import org.bukkit.command.RemoteConsoleCommandSender;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.server.ServerLoadEvent;
+import org.bukkit.plugin.java.JavaPlugin;
+
+public abstract class CommandAPISpigot<Source> extends CommandAPIBukkit<Source> implements SpigotNMS<Source> {
+
+	private static CommandAPISpigot<?> spigot;
+
+	@SuppressWarnings("unchecked")
+	protected CommandAPISpigot() {
+		this.nms = (NMS<Source>) bukkitNMS();
+		CommandAPISpigot.spigot = this;
+		setInstance(this);
+	}
+
+	@SuppressWarnings("unchecked")
+	public static <Source> CommandAPISpigot<Source> getSpigot() {
+		return (CommandAPISpigot<Source>) spigot;
```

Comment:
Should this have a `hey, you're calling this too early` check like there is in [`CommandAPIBukkit`](https://github.com/JorelAli/CommandAPI/blob/7d2a622ab856eabf2393d8c35bee309bad646177/commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/CommandAPIBukkit.java#L68)?